packages/core/src/browser/label-provider.ts

```diff
@@ -38,6 +38,10 @@ export interface LabelProviderContribution {
      */
     getLongName?(element: object): string;
 
+    /**
+     * returns a directory name for the given element.
+     */
+    getDirectoryName?(element: object): string;
```

Comment:
could we remove it? instead you should ask for parent uri of you uri and call `getLongName`

Reply:
label provider is not specific to uris and should not be aware about uri specific concepts

Reply:
Ah indeed, that makes much more sense.